### PR TITLE
Backport of hotfixes #10538 and #10605 

### PR DESCRIPTION
**Pull request by bkase**
Backporting  #10538 and #10605 over to compatible instead

Add `MINA_ROSETTA_TERMINATE_ON_SERVER_ERROR` feature.

### CODE REVIEW

**Review by lk86 — APPROVED** [review]

File: src/app/rosetta/lib/account.ml [hunk 1]
```diff
@@ -264,17 +373,22 @@ AS combo GROUP BY combo.pk_id
...
-        | Some ((_,_,Some _) as result) -> return @@ Some result
-        | Some (_,_,None)
+        | Some ((_pk,slot,balance,Some nonce)) ->
+            return @@ Some (slot, balance, Some nonce)
+        | Some (_,_,_,None)
         | None ->
-          Conn.find_opt query_canonical_fallback (address, requested_block_height))
+          let%map result = Conn.find_opt query_canonical_fallback (address, requested_block_height) in
+          (* The nonce is returned from this user-command, so we need to add one from here to get the current nonce in the account. *)
+          Option.map result ~f:(fun (slot,balance,nonce) -> (slot,balance,Option.map ~f:Int64.((+) one) nonce)))
```
Comment (psteckler) [inline src/app/rosetta/lib/account.ml:382]: can you use `succ`?

**Review by mrmr1993 — APPROVED** [review]
Nice

**Review by psteckler — APPROVED** [review]
one nit

File: src/app/rosetta/lib/account.ml [hunk 2]
```diff
@@ -29,14 +29,36 @@ module Balance_info = struct
...
+  MAX(combo.balance) AS balance,
+  MAX(combo.nonce) AS nonce
+FROM (
+  /* There are two large recursive subqueries here. One for balance, and the
+   * other for nonce.
+   *
+   * These are separate subqueries because there is a quirk where a transaction
+   * can be received and sent in the same block and the latest nonce is not
+   * quite stored properly. To work around that, we are going to query _almost_
+   * the same thing but take the MAX nonce among the most recent 255 entries
```
Comment (psteckler) [inline src/app/rosetta/lib/account.ml:56]: probably shouldn't mention the limit, even in a comment, since it may later change

**Comment by bkase** [comment]
!ci-build-me

**Review by bkase — COMMENTED** [review]
I can't approve my own pull request, but consider this approving @psteckler 's e25a67940a0ff9b7461

**Comment by joseandro** [comment]
For some reason this PR also showed up "In progress" in the Velocity board. I am updating it back to "New Issues" in that board. This action shouldn't affect the original position/state of this issue in its original board.

**Comment by mrmr1993** [comment]
Stale, closing.

**Pull request closed without merging**